app/index.js

```diff
@@ -64,17 +65,23 @@ module.exports = yeoman.generators.Base.extend({
   loadTemplates: function() {
     var done = this.async();
 
-    Workspace.getAvailableTemplates(function(err, list) {
+    Workspace.describeAvailableTemplates(function(err, list) {
       if (err) return done(err);
       this.templates = list.map(function(t) {
         return {
-          // TODO(bajtos) - workspace does not provide template details yet
-          // name: util.format('%s (%s)', t.name, t.description),
-          // value: t.name
-          name: t,
-          value: t
+          name: util.format('%s (%s)', t.name, t.description),
+          value: t.name
         };
       });
+
+      if (helpers.getCommandName() === 'apic') {
```

Comment:
I think https://github.com/strongloop/generator-loopback/issues/139 will enable us to get rid of this easily. I'll leave a TODO comment.